### ISSUE TITLE
vim's wildmode completion setting

### ISSUE THREAD

**Issue by aleprovencio** [qutebrowser/qutebrowser#3517]
Hello! Is there a way to change the way the completion system work to something like vim's wildmode config? My favorite is the popular `list:longest,full`.

**Comment by The-Compiler** [comment]
Can you elaborate on what behavior you want exactly?

**Comment by mschilli87** [comment]
You can see an example on how to use it in vim on [reddit](https://www.reddit.com/r/vim/comments/19izuz/whats_your_wildmode/).

Essentially it boils down to:

 1. open `vim`
 2. run `:set wildmode=longest:list,full`
 3. run `:set wildmenu`
 4. type `:echo <Tab>`

**Comment by The-Compiler** [comment]
I know the wildmode setting, I'm still wondering how that'd map to qutebrowser's completion exactly :wink:

**Comment by aleprovencio** [comment]
Say we want to achieve `:config-source` from the command line:

* Type `:co` and hit `<tab>`
* The string would be completed to the longest common string (i.e. `:config-`) 
* A list of the avaiable options would be shown on the menu
* These options are chosen by the subsequent `<tab>` hits.

**Comment by The-Compiler** [comment]
Thanks for the explanation! I'm not so sure about this yet as the completion code is already quite complex. But I agree it'd probably be useful. cc @rcorre

**Comment by rcorre** [comment]
It would have to be configurable per-completion, as you probably wouldn't want it for url completion. I do see it being useful for `command`/`config` completion, if it doesn't end up being too complex

**Comment by aleprovencio** [comment]
Yes @rcorre, indeed I was planning it for `command/config completion`. I guess even `URL completion` can also be improved but in a different manner on another issue, so it makes sense to have different configurations for those.
